pkg/sources/git/git.go

```diff
@@ -453,6 +458,55 @@ func (s *Git) ScanRepo(ctx context.Context, repo *git.Repository, repoPath strin
 	return nil
 }
 
+func normalizeConfig(scanOptions *ScanOptions, repo *git.Repository) (err error) {
```

Comment:
**Suggestion (optional):** We may be able to minimize our if statements by starting the function with something like:

```go
if scanOptions.BaseHash == "" || scanOptions.HeadHash == "" {
    // Nothing to do.
    return nil
}

// ...
```